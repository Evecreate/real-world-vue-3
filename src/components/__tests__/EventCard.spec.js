import { describe, it, expect } from 'vitest';

import { shallowMount } from '@vue/test-utils';
import EventCard from '../EventCard.vue';

describe('EventCard', () => {
  it('renders properly', () => {
    const wrapper = shallowMount(EventCard, {
      props: {
        event: {
          id: 123,
          category: 'animal welfare',
          title: 'Cat Adoption Day',
          description: 'Find your new feline friend at this event.',
          location: 'Meow Town',
          date: 'January 28, 2022',
          time: '12:00',
          petsAllowed: true,
          organizer: 'Kat Laydee',
        },
      },
      stubs: ['RouterLink']
    });
    expect(wrapper.html()).toMatchSnapshot();
  });
});
