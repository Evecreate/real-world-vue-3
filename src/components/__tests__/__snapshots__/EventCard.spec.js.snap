// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`EventCard > renders properly 1`] = `
"<routerlink data-v-f2df5c1a="" to="[object Object]" class="event-link">
  <div data-v-f2df5c1a="" class="event-card">
    <h2 data-v-f2df5c1a="">Cat Adoption Day</h2><span data-v-f2df5c1a="">@12:00 on January 28, 2022</span>
  </div>
</routerlink>"
`;
